fix(CookieChangeEvent): revise misleading descriptions

xyy94813 opened the pull request:
<!-- 🙌 Thanks for contributing to MDN Web Docs. Adding details below will help us to merge your PR faster. -->

### Description

<!-- ✍️ Summarize your changes in one or two sentences -->

- Emphasizes the existence of `cookie changes`.
- Emphasizes that `type` belongs to `cookie changes`.

### Motivation

<!-- ❓ Why are you making these changes and how do they help readers? -->
The previous description misinterpreted `type` as referring to the type attribute of the CookieChangeEvent.

### Additional details

<!-- 🔗 Link to release notes, vendor docs, bug trackers, source control, or other places providing more context -->
https://wicg.github.io/cookie-store/#process-changes

### Related issues and pull requests

<!-- 🔨 If this fully resolves a GitHub issue, use "Fixes #123" -->
<!-- 👉 Highlight related pull requests using "Relates to #123" -->
<!-- ❗ If another pull request should be merged first, use "**Depends on:** #123" -->
Relates to https://github.com/mdn/translated-content/pull/19845

<!-- 👷‍♀️ After submitting, go to the "Checks" tab of your PR for the build status -->

xyy94813 committed:
fix(CookieChangeEvent): revise misleading descriptions

skyclouds2001 commented on files/en-us/web/api/cookiechangeevent/index.md:
```diff
@@ -9,13 +9,12 @@ browser-compat: api.CookieChangeEvent
-
-> **Note:** A cookie that is replaced due to the insertion of another cookie with the same name, domain, and path, is ignored and does not trigger a change event.
+- A cookie which is removed due to an insertion of another cookie with the same name, domain, and path is ignored.
+- A newly-created cookie which is not immediately evicted. In this case `type` of cookie-change is "changed".
+- A newly-created cookie which is immediately evicted. In this case `type` of cookie-change is "deleted".
+- A cookie which is otherwise evicted or removed. In this case `type` of cookie-change is "deleted".
```
```suggestion
- A newly-created cookie which is not immediately evicted. In this case `type` of cookie-change is `"changed"`.
- A newly-created cookie which is immediately evicted. In this case `type` of cookie-change is `"deleted"`.
- A cookie which is otherwise evicted or removed. In this case `type` of cookie-change is `"deleted"`.
```

xyy94813 replied:
The current problem is not at the formatting, resolve the suggestion for now.

wbamberg reviewed (commented):
The original page doesn't make sense but this PR doesn't make sense either. It seems to me like we want to say something like:

```
The `CookieChangeEvent` interface of the {{domxref("Cookie Store API", "", "", "nocode")}}represents a change to the contents of the cookie store. An instance of a `CookieChangeEvent` is passed into the event handler for the {{domxref("CookieStore.change_event", "change")}} event of the `CookieStore` interface.

The `CookieChangeEvent` interface has two properties to represent cookie changes:

- `changed`: an array of objects representing cookies that have been newly created and not immediately evicted.
- `deleted`: an array of objects representing cookies which have been newly created and immediately evicted, or cookies which have been otherwise evicted or removed.

Note that the `change` event ignores cookies which were removed removed due to an insertion of another cookie with the same name, domain, and path.
```

xyy94813 commented:
After verification, I have question about "immediately evicted". 

See the follow test case:

```js
cookieStore.addEventListener("change", (event) => {
  console.log(event)
});
// Test Case 1: set  and delete immediately 
// actually, log twice: 
// firstly, log CookieChangeEvent { changed(1), deleted(0) }
// seccondly, log CookieChangeEvent { changed(0), deleted(1) }
cookieStore.set('cookie1', '123');
cookieStore.delete('cookie1');

// Test Case 2:  set  and delete immediately 2
// actually, log twice: 
// firstly, log CookieChangeEvent { changed(1), deleted(0) }
// seccondly, log CookieChangeEvent { changed(0), deleted(1) }
const cookie3 = {
  name: 'c2',
  value: 'c2222',
}
cookieStore.set(cookie3 );
cookieStore.delete(cookie3);

// Test Case 3: set a expire cookie (passed)
// actually log once:  CookieChangeEvent { changed(0), deleted(1) }
cookieStore.set({
  name: 'c2',
  value: 'c2222',
  expire: Date.now() - 100,
});
```

How to define "immediately evicted" ?
Is Test Case1 and Test Case2 passed?

xyy94813 commented:
> The original page doesn't make sense but this PR doesn't make sense either. It seems to me like we want to say something like:
> 
> ```
> The `CookieChangeEvent` interface of the {{domxref("Cookie Store API", "", "", "nocode")}}represents a change to the contents of the cookie store. An instance of a `CookieChangeEvent` is passed into the event handler for the {{domxref("CookieStore.change_event", "change")}} event of the `CookieStore` interface.
> 
> The `CookieChangeEvent` interface has two properties to represent cookie changes:
> 
> - `changed`: an array of objects representing cookies that have been newly created and not immediately evicted.
> - `deleted`: an array of objects representing cookies which have been newly created and immediately evicted, or cookies which have been otherwise evicted or removed.
> 
> Note that the `change` event ignores cookies which were removed removed due to an insertion of another cookie with the same name, domain, and path.
> ```

Wouldn't it be better to describe the details in the properties section?

xyy94813 committed:
Merge branch 'main' of https://github.com/mdn/content into hotfix/cookie-change-event

xyy94813 committed:
fix(CookieChangeEvent): revise misleading descriptions

skyclouds2001 commented on files/en-us/web/api/cookiechangeevent/index.md:
```diff
@@ -7,15 +7,15 @@ browser-compat: api.CookieChangeEvent
+- A cookie which is otherwise evicted or removed is considered "deleted".
 
-> **Note:** A cookie that is replaced due to the insertion of another cookie with the same name, domain, and path, is ignored and does not trigger a change event.
+> **Note:** that the `change` event ignores cookies which were removed removed due to an insertion of another cookie with the same name, domain, and path.
```
```suggestion
> **Note:** that the `change` event ignores cookies which were removed due to an insertion of another cookie with the same name, domain, and path.
```

xyy94813 committed:
Update files/en-us/web/api/cookiechangeevent/index.md

Co-authored-by: skyclouds2001 <95597335+skyclouds2001@users.noreply.github.com>

wbamberg commented:
> After verification, I have question about "immediately evicted".

Thanks for raising this! I've asked for clarification in https://github.com/WICG/cookie-store/issues/229.

wbamberg commented:
> > The original page doesn't make sense but this PR doesn't make sense either. It seems to me like we want to say something like:
> > ```
> > The `CookieChangeEvent` interface of the {{domxref("Cookie Store API", "", "", "nocode")}}represents a change to the contents of the cookie store. An instance of a `CookieChangeEvent` is passed into the event handler for the {{domxref("CookieStore.change_event", "change")}} event of the `CookieStore` interface.
> > 
> > The `CookieChangeEvent` interface has two properties to represent cookie changes:
> > 
> > - `changed`: an array of objects representing cookies that have been newly created and not immediately evicted.
> > - `deleted`: an array of objects representing cookies which have been newly created and immediately evicted, or cookies which have been otherwise evicted or removed.
> > 
> > Note that the `change` event ignores cookies which were removed removed due to an insertion of another cookie with the same name, domain, and path.
> > ```
> 
> Wouldn't it be better to describe the details in the properties section?

Yes, probably. You mean something like:

```
- changed
  - : An array listing all newly-created cookies which were not immediately evicted.
- deleted
  - : An array listing all newly-created cookies which were immediately evicted, and all cookies which were otherwise evicted or removed.
```

?

xyy94813 commented:
> > > The original page doesn't make sense but this PR doesn't make sense either. It seems to me like we want to say something like:
> > > ```
> > > The `CookieChangeEvent` interface of the {{domxref("Cookie Store API", "", "", "nocode")}}represents a change to the contents of the cookie store. An instance of a `CookieChangeEvent` is passed into the event handler for the {{domxref("CookieStore.change_event", "change")}} event of the `CookieStore` interface.
> > > 
> > > The `CookieChangeEvent` interface has two properties to represent cookie changes:
> > > 
> > > - `changed`: an array of objects representing cookies that have been newly created and not immediately evicted.
> > > - `deleted`: an array of objects representing cookies which have been newly created and immediately evicted, or cookies which have been otherwise evicted or removed.
> > > 
> > > Note that the `change` event ignores cookies which were removed removed due to an insertion of another cookie with the same name, domain, and path.
> > > ```
> > 
> > 
> > Wouldn't it be better to describe the details in the properties section?
> 
> Yes, probably. You mean something like:
> 
> ```
> - changed
>   - : An array listing all newly-created cookies which were not immediately evicted.
> - deleted
>   - : An array listing all newly-created cookies which were immediately evicted, and all cookies which were otherwise evicted or removed.
> ```
> 
> ?

Yes. 

```
...

## Instance properties

_This interface also inherits properties from {{domxref("Event")}}._
- {{domxref("CookieChangeEvent.changed")}} {{ReadOnlyInline}}
  - : Returns an an array of objects representing cookies that have been newly created and not immediately evicted.
- {{domxref("CookieChangeEvent.deleted")}} {{ReadOnlyInline}}
  - : `deleted`: an array of objects representing cookies which have been newly created and immediately evicted, or cookies which have been otherwise evicted or removed.
```

wbamberg commented:
Based on https://github.com/WICG/cookie-store/issues/229, I wonder if it would be better to remove the "immediately evicted" language from MDN, since that seems to be more targeted at people implementing the API, rather than people using it.

something like:

```
- changed
  - : An array listing all newly-created cookies. Note that this will exclude cookies which were set with an expiry date in the past.
- deleted
  - : An array listing all cookies which were removed, either because they expired or because they were explicitly deleted. Note that this will include cookies which were set with an expiry date in the past.
```

? We could also keep an eye on https://github.com/WICG/cookie-store/issues/229 and revise this again if it seems to be needed...

Note that we do already say, in https://developer.mozilla.org/en-US/docs/Web/API/CookieStore/delete:

> The delete() method expires the cookie by changing the date to one in the past.

wbamberg commented:
@xyy94813 , please let me know if you are coming back to this, if not I'm happy to try to finish it off.

xyy94813 commented:
No.
I haven't had enough time to focus on this issue recently.
